File: src/main/java/com/youngjun/auth/core/domain/auth/AuthService.java
package com.youngjun.auth.core.domain.auth;

import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.stereotype.Service;

@Service
public class AuthService implements UserDetailsService {

    private final AuthReader authReader;

    private final AuthWriter authWriter;

    public AuthService(AuthReader authReader, AuthWriter authWriter) {
        this.authReader = authReader;
        this.authWriter = authWriter;
    }

    @Override
    public Auth loadUserByUsername(String username) {
        return authReader.read(username);
    }

    public Auth register(NewAuth newAuth) {
        return authWriter.write(newAuth);
    }

}

File: src/main/kotlin/com/youngjun/auth/core/api/application/AuthService.kt
package com.youngjun.auth.core.api.application

import com.youngjun.auth.core.domain.auth.Auth
import com.youngjun.auth.core.domain.auth.AuthWriter
import com.youngjun.auth.core.domain.auth.NewAuth
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
class AuthService(
    private val authWriter: AuthWriter,
    private val passwordEncoder: PasswordEncoder,
) {
    fun register(newAuth: NewAuth): Auth = authWriter.write(newAuth.encodedWith(passwordEncoder))
}
